HandleNHDeleteInstallationCallOrchestrator/handler.ts

```diff
@@ -24,51 +32,143 @@ export type NhDeleteInstallationOrchestratorCallInput = t.TypeOf<
   typeof NhDeleteInstallationOrchestratorCallInput
 >;
 
+export type OrchestratorSuccess = t.TypeOf<typeof OrchestratorSuccess>;
+export const OrchestratorSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+
+export type OrchestratorInvalidInputFailure = t.TypeOf<
+  typeof OrchestratorInvalidInputFailure
+>;
+export const OrchestratorInvalidInputFailure = t.interface({
+  input: t.unknown,
+  kind: t.literal("FAILURE_INVALID_INPUT"),
+  reason: t.string
+});
+
+export type OrchestratorActivityFailure = t.TypeOf<
+  typeof OrchestratorActivityFailure
+>;
+export const OrchestratorActivityFailure = t.interface({
+  activityName: t.string,
+  kind: t.literal("FAILURE_ACTIVITY"),
+  reason: t.string
+});
+
+export type OrchestratorUnhandledFailure = t.TypeOf<
+  typeof OrchestratorUnhandledFailure
+>;
+export const OrchestratorUnhandledFailure = t.interface({
+  kind: t.literal("FAILURE_UNHANDLED"),
+  reason: t.string
+});
+
+export type OrchestratorFailure = t.TypeOf<typeof OrchestratorFailure>;
+export const OrchestratorFailure = t.union([
+  OrchestratorActivityFailure,
+  OrchestratorInvalidInputFailure,
+  OrchestratorUnhandledFailure
+]);
+
```

Comment:
Do you think we can move this types to a separate file, so we can share them between different orchestrators?

Reply:
I feel the same